extractor/docker/docker_test.go

```diff
@@ -211,11 +355,13 @@ func TestApplyLayers(t *testing.T) {
 	}
 
 	for _, tc := range testCases {
-		gotImageDetail := ApplyLayers(tc.inputLayers)
-		sort.Slice(gotImageDetail.Packages, func(i, j int) bool {
-			return gotImageDetail.Packages[i].Name < gotImageDetail.Packages[j].Name
+		t.Run(tc.name, func(t *testing.T) {
+			gotImageDetail := ApplyLayers(tc.inputLayers)
+			sort.Slice(gotImageDetail.Packages, func(i, j int) bool {
+				return gotImageDetail.Packages[i].Name < gotImageDetail.Packages[j].Name
+			})
```

Comment:
This sorting function doesn't always work. Here's an example run:

```
  --- FAIL: TestApplyLayers/happy_path_with_removed_and_updated_lockfile (0.00s)
        docker_test.go:363: 
            	Error Trace:	docker_test.go:363
            	Error:      	Not equal: 
            	            	expected: types.ImageDetail{OS:(*types.OS)(0xc0002e8cc0), Packages:[]types.Package(nil), Applications:[]types.Application{types.Application{Type:"gem", FilePath:"app/Gemfile.lock", Libraries:[]types.LibraryInfo{types.LibraryInfo{Library:types.Library{Name:"rails", Version:"6.0.0"}, LayerID:"sha256:24df0d4e20c0f42d3703bf1f1db2bdd77346c7956f74f423603d651e8e5ae8a7"}, types.LibraryInfo{Library:types.Library{Name:"rack", Version:"4.0.0"}, LayerID:"sha256:932da51564135c98a49a34a193d6cd363d8fa4184d957fde16c9d8527b3f3b02"}}}, types.Application{Type:"composer", FilePath:"app/composer2.lock", Libraries:[]types.LibraryInfo{types.LibraryInfo{Library:types.Library{Name:"phplibrary1", Version:"6.6.6"}, LayerID:"sha256:24df0d4e20c0f42d3703bf1f1db2bdd77346c7956f74f423603d651e8e5ae8a7"}}}}, HistoryPackages:[]types.Package(nil)}
            	            	actual  : types.ImageDetail{OS:(*types.OS)(0xc0002e8ca0), Packages:[]types.Package(nil), Applications:[]types.Application{types.Application{Type:"composer", FilePath:"app/composer2.lock", Libraries:[]types.LibraryInfo{types.LibraryInfo{Library:types.Library{Name:"phplibrary1", Version:"6.6.6"}, LayerID:"sha256:24df0d4e20c0f42d3703bf1f1db2bdd77346c7956f74f423603d651e8e5ae8a7"}}}, types.Application{Type:"gem", FilePath:"app/Gemfile.lock", Libraries:[]types.LibraryInfo{types.LibraryInfo{Library:types.Library{Name:"rails", Version:"6.0.0"}, LayerID:"sha256:24df0d4e20c0f42d3703bf1f1db2bdd77346c7956f74f423603d651e8e5ae8a7"}, types.LibraryInfo{Library:types.Library{Name:"rack", Version:"4.0.0"}, LayerID:"sha256:932da51564135c98a49a34a193d6cd363d8fa4184d957fde16c9d8527b3f3b02"}}}}, HistoryPackages:[]types.Package(nil)}
            	            	
            	            	Diff:
            	            	--- Expected
            	            	+++ Actual
            	            	@@ -7,2 +7,15 @@
            	            	  Applications: ([]types.Application) (len=2) {
            	            	+  (types.Application) {
            	            	+   Type: (string) (len=8) "composer",
            	            	+   FilePath: (string) (len=18) "app/composer2.lock",
            	            	+   Libraries: ([]types.LibraryInfo) (len=1) {
            	            	+    (types.LibraryInfo) {
            	            	+     Library: (types.Library) {
            	            	+      Name: (string) (len=11) "phplibrary1",
            	            	+      Version: (string) (len=5) "6.6.6"
            	            	+     },
            	            	+     LayerID: (digest.Digest) (len=71) sha256:24df0d4e20c0f42d3703bf1f1db2bdd77346c7956f74f423603d651e8e5ae8a7
            	            	+    }
            	            	+   }
            	            	+  },
            	            	   (types.Application) {
            	            	@@ -26,15 +39,2 @@
            	            	    }
            	            	-  },
            	            	-  (types.Application) {
            	            	-   Type: (string) (len=8) "composer",
            	            	-   FilePath: (string) (len=18) "app/composer2.lock",
            	            	-   Libraries: ([]types.LibraryInfo) (len=1) {
            	            	-    (types.LibraryInfo) {
            	            	-     Library: (types.Library) {
            	            	-      Name: (string) (len=11) "phplibrary1",
            	            	-      Version: (string) (len=5) "6.6.6"
            	            	-     },
            	            	-     LayerID: (digest.Digest) (len=71) sha256:24df0d4e20c0f42d3703bf1f1db2bdd77346c7956f74f423603d651e8e5ae8a7
            	            	-    }
            	            	-   }
            	            	   }
            	Test:       	TestApplyLayers/happy_path_with_removed_and_updated_lockfile
            	Messages:   	happy path with removed and updated lockfile
```

Reply:
Thanks! I've fixed it. https://github.com/aquasecurity/fanal/pull/93/commits/8c8afacf9b20eeb98a3e1e430aede465515c864e